inc/dashboard/filter.class.php

```diff
@@ -192,17 +192,17 @@ static function displayList(
       $js = <<<JAVASCRIPT
       var on_change_{$rand} = function() {
          var dom_elem    = $('#dropdown_{$fieldname}{$rand}');
-         var selected    = parseInt(dom_elem.find(':selected').val());
+         var selected    = dom_elem.find(':selected').val();
 
          Dashboard.saveFilter('{$fieldname}', selected);
 
-         $(dom_elem).closest("fieldset").toggleClass("filled", selected > 0)
+         $(dom_elem).closest("fieldset").toggleClass("filled", selected !== null)
```

Comment:
when a filter is set, the value of the input is null. If it is possible to have a string value, null is better to use for this check